observability/observability.go

```diff
@@ -173,12 +174,19 @@ func RecordMetricsForMetricsExporter(ctx context.Context, receivedTimeSeries int
 }
 
 // MakeComponentName constructs the name of the component (for the moment receiver and exporter) that should be used for observability.
-// E.g. as a tag/metrics label or span name.
-func MakeComponentName(typeStr, name string) string {
-	if len(name) == 0 {
+// E.g. as a tag/metrics label or span name. The normalized fullName must use the format "typeStr/name", the result of this should be "typeStr_name".
+func MakeComponentName(typeStr, fullName string) string {
+	// The fullName is not normalized, use typeStr
+	if typeStr != fullName && !strings.HasPrefix(fullName, typeStr+config.TypeAndNameSeparator) {
```

Comment:
I am not sure I understand what case this is. Do we have a case where `fullName` is some arbitrary string?